src/contexts/Apollo.tsx

```diff
@@ -19,22 +30,103 @@ interface Props {
 export const ApolloProviderWrapper = ({ children }: Props) => {
   const { status } = useWeb3Context()
 
-  const client = React.useMemo(() => {
-    let httpUri = GRAPH_HTTP_MAINNET
-    let wsUri = GRAPH_WS_MAINNET
+  const OmenLink = React.useMemo(() => {
+    let httpUri = OMEN_GRAPH_HTTP_MAINNET
+    let wsUri = OMEN_GRAPH_WS_MAINNET
 
     if (DEFAULT_NETWORK_ID === (4 as NetworkId)) {
-      httpUri = GRAPH_HTTP_RINKEBY
-      wsUri = GRAPH_WS_RINKEBY
+      httpUri = OMEN_GRAPH_HTTP_RINKEBY
+      wsUri = OMEN_GRAPH_WS_RINKEBY
     }
 
     if (status._type === Web3ContextStatus.Connected) {
       const { networkConfig } = status
-      ;({ httpUri, wsUri } = networkConfig.getGraphUris())
+      const { OMENhttpUri, OMENwsUri } = networkConfig.getGraphUris()
+      httpUri = OMENhttpUri
+      wsUri = OMENwsUri
     }
 
-    return getApolloClient(httpUri, wsUri)
+    const OmenWsLink = new WebSocketLink({
+      uri: wsUri,
+      options: {
+        reconnect: true,
+      },
+    })
+
+    const OmenHttpLink = new HttpLink({
+      uri: httpUri,
+    })
+
+    return split(
+      ({ query }) => {
+        const definition = getMainDefinition(query)
+        return definition.kind === 'OperationDefinition' && definition.operation === 'subscription'
+      },
+      OmenWsLink,
+      OmenHttpLink
+    )
+  }, [status])
+
+  const CTELink = React.useMemo(() => {
+    let httpUri = CTE_GRAPH_HTTP_MAINNET
+    let wsUri = CTE_GRAPH_WS_MAINNET
+
+    if (DEFAULT_NETWORK_ID === (4 as NetworkId)) {
+      httpUri = CTE_GRAPH_HTTP_RINKEBY
+      wsUri = CTE_GRAPH_WS_RINKEBY
+    }
+
+    if (status._type === Web3ContextStatus.Connected) {
+      const { networkConfig } = status
+      const { CTEhttpUri, CTEwsUri } = networkConfig.getGraphUris()
+      httpUri = CTEhttpUri
+      wsUri = CTEwsUri
+    }
+
+    const CTEWsLink = new WebSocketLink({
+      uri: wsUri,
+      options: {
+        reconnect: true,
+      },
+    })
+
+    const CTEHttpLink = new HttpLink({
+      uri: httpUri,
+    })
+
+    return split(
+      ({ query }) => {
+        const definition = getMainDefinition(query)
+        return definition.kind === 'OperationDefinition' && definition.operation === 'subscription'
+      },
```

Comment:
I'm not completely sure about this, but that predicate is twice on the code, my understanding is that is for splitting the link between websocket operations an no-websocket, so maybe a named function predicate that returns this could clarify the purpose of those lines.

Reply:
Done